io_u.c

```diff
@@ -884,6 +883,24 @@ static int fill_io_u(struct thread_data *td, struct io_u *io_u)
 	}
 
 	/*
+	 * If zone_size > zone_range, then maintain the same zone until
+	 * zone_bytes >= zone_size.
+	 */
+	if (f->last_pos[io_u->ddir] >= (f->file_offset + td->o.zone_range)) {
+		dprint(FD_IO, "io_u maintain zone offset=%" PRIu64 "/last_pos=%" PRIu64 "\n",
+				f->file_offset, f->last_pos[io_u->ddir]);
+		f->last_pos[io_u->ddir] = f->file_offset;
+	}
+
+	/*
+	 * For random: if 'norandommap' is not set and zone_size > zone_range,
+	 * map needs to be reset as it's done with zone_range everytime.
+	 */
+	if (td->o.zone_range && (td->zone_bytes % td->o.zone_range == 0)) {
+		fio_file_reset(td, f);
```

Comment:
You might want to add the extra parenthesis Jens mentioned here - it might not be obvious that % will bind tighter than == .